hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/ActionRepository.java

```diff
@@ -352,6 +352,22 @@ Long countByRolloutAndRolloutGroupAndStatusNotAndStatusNotAndStatusNot(JpaRollou
             JpaRolloutGroup rolloutGroupParent, Status actionStatus);
 
     /**
+     * Retrieving all actions referring to a given rollout group with a specific
+     * a specific status.
+     *
+     * @param rollout
+     *            the rollout the actions belong to
+     * @param rolloutGroupParent
+     *            the parent rollout group the actions should reference
+     * @param actionStatus
+     *            the status the actions have
+     * @return the actions referring a specific rollout group and a specific
+     *         status
+     */
+    List<Identifiable<Long>> findByRolloutAndRolloutGroupAndStatus(JpaRollout rollout,
```

Comment:
Finally this is fed to `DeploymentManagament#startScheduledAction`, which only accepts the `actionId`. So this is a memory optimization. It takes less memory to have a list of thousands of Longs than thousands of Actions.
I also just copied it from `findByRolloutAndRolloutGroupParentAndStatus`